### ISSUE TITLE
Unable to remove treatment

### ISSUE THREAD

**Issue by st8ass** [nightscout/cgm-remote-monitor#6014]
After migration to the Atlas and updating NS to 14.0.3 version, I can't remove treatment. I try to delete, but nothing happens

NS 14.0.3
Freestyle Libre 1/Miao-Miao
Chrome (version 85)

![Screenshot 2020-09-16 191912](https://user-images.githubusercontent.com/36138923/93369117-8f67a900-f857-11ea-93ed-92da245bc358.png)

**Comment by Batista1991** [comment]
yep, have same problem on atlas + new NS version

**Comment by Anatoliy91** [comment]
I have same problem from 7.08.2020, please fix it

**Comment by bjornoleh** [comment]
I have also noticed that careportal entries are not updated properly. If you go to Heroku and do a "Restart all dynos", the careportal entries are updated correctly. Some observations of mine:

-Entries at the current time are OK
-Entries of past events need restart of dynos
-Deleting careportal entries need restart of dynos

**Comment by sulkaharo** [comment]
Ok cause know, working on a fix for the .4 release.

**Comment by dicko72** [comment]
The treatment deletions do trigger the update in treatments table within collections on Atlas. However they remain on the UI.

**Comment by bjornoleh** [comment]
@dicko72 Same with mLab. In a pinch, it's possible to do the "Restart all dynos" trick while we are waiting for the fix.

**Comment by sulkaharo** [comment]
I have a preliminary fix but it needs some more effort. I'll post a branch for testing soon

**Comment by sulkaharo** [comment]
#6056 should fix this - if you want to help test, you'll need to deploy branch cache_invalidation to your Heroku setup

**Comment by bjornoleh** [comment]
@sulkaharo Ok, I tried now, but unfortunately it did not seem to make a difference. I am still in mLab, should that make a difference?

I could have made some error, but here is the branch I deployed (I made a copy of Master 13.0.1 and made a PR to your testing branch):
https://github.com/bjornoleh/cgm-remote-monitor/compare/cache_invalidation...nightscout:cache_invalidation

This is what I tested:

Add "Carb correction" with Careportal at 12:30 h
Confirmed entry in mLab
Remove entry
Confirmed removal in mLab
Entry still visible in NS, also after ctrl-f5

Needed Restart all dynos to clear entry from NS

**Comment by st8ass** [comment]
> #6056 should fix this - if you want to help test, you'll need to deploy branch cache_invalidation to your Heroku setup

Unfortunately, it didn't solve the problem, also ifttt and automagic buttons stopped working - I can't enter temp targets and carbs

**Comment by sulkaharo** [comment]
@bjornoleh how are you removing the data? I double checked and edits that use the rest api work as expected (such as using the Treatments report). Direct data editing in Mongo is not supported, so if you're making changes to the data directly in the database, that won't work.

**Comment by sulkaharo** [comment]
@st8ass can you double check you're in the right branch and describe how you tested? If your ifft action doesn't work, double check you're sending the required created_at treatment date field that contains an ISO-8601 date

**Comment by bjornoleh** [comment]
@sulkaharo 
I removed the treatment usingvthe edit function in NS, drag and drop to the delete field to the left. I just checked in mLab to verify that the treatment was entered and removed, which it was.

**Comment by st8ass** [comment]
> @st8ass can you double check you're in the right branch and describe how you tested? If your ifft action doesn't work, double check you're sending the required created_at treatment date field that contains an ISO-8601 date

yes, I selected the cache_invalidation branch
![image](https://user-images.githubusercontent.com/36138923/93720014-0d48ee80-fb8f-11ea-8f5d-156f02371c5e.png)

**Comment by st8ass** [comment]
ifttt buttons work correctly on the master branch

Example of applet settings
![smb](https://user-images.githubusercontent.com/36138923/93720802-0a9cc800-fb94-11ea-92b3-a204419f746e.jpg)

**Comment by sulkaharo** [comment]
Right, I found an issue with the treatment edit mode in the UI, fixing.

@st8ass The body is missing a required filed -> try adding  _"created_at": {{OccurredAt}}_  to the JSON.

**Comment by sulkaharo** [comment]
Ok branch has support for the edit mode now. I removed the data validation from the API so IFTTT calls etc that send data with missing dates will work again.

**Comment by bjornoleh** [comment]
@sulkaharo Cool, it works for me, I tested entering, moving and deleting a Carb correction entry.

**Comment by st8ass** [comment]
@sulkaharo IFTTT buttons and removing treatment works correctly 
Thanks for your hard work. I am very grateful to you for developing and supporting Nightscout

**Comment by dicko72** [comment]
> > @st8ass can you double check you're in the right branch and describe how you tested? If your ifft action doesn't work, double check you're sending the required created_at treatment date field that contains an ISO-8601 date
> 
> yes, I selected the cache_invalidation branch
> ![image](https://user-images.githubusercontent.com/36138923/93720014-0d48ee80-fb8f-11ea-8f5d-156f02371c5e.png)
> 
> I removing the treatments using the edit function in NS, just like the screenshot at the top of the discussion

How do I get this branch to show up in Heroku?

**Comment by st8ass** [comment]
> > > @st8ass can you double check you're in the right branch and describe how you tested? If your ifft action doesn't work, double check you're sending the required created_at treatment date field that contains an ISO-8601 date
> > 
> > 
> > yes, I selected the cache_invalidation branch
> > ![image](https://user-images.githubusercontent.com/36138923/93720014-0d48ee80-fb8f-11ea-8f5d-156f02371c5e.png)
> > I removing the treatments using the edit function in NS, just like the screenshot at the top of the discussion
> 
> How do I get this branch to show up in Heroku?

1. Make a Fork of Nightscout/cgm-remote-monitor to your own repository. 
2. Login to Heroku. Open deploy tab and connect Heroku to your Github account. 
3. Select the Nightscout repository, scroll down to Manual deploy and select the cache_invalidation branch. 
4. Click the Deploy Branch button and wait for the deployment to complete

Done )

Sorry if I didn't explain it well, I'm not a developer

**Comment by bjornoleh** [comment]
Anyone with a NS site will already have a fork of NS in their Github
account. It is possible to delete the fork and make a new one to get access
to newly added branches, like the one we are testing now.

But there are other options. I have just found one way, and would be happy
to learn better ways of doing this.

I made a copy of one of the other branches, and named it cach_invalidation
(the name doesn't matter). Then I made a pull request from the original
branch in the NS repo.

Any better suggestions?

man. 21. sep. 2020, 23:14 skrev st8ass <notifications@github.com>:
